pkgs/top-level/go-packages.nix

```diff
@@ -2671,13 +2665,21 @@ let
       pushover
     ];
 
-    buildFlagsArray = ''
+    # Tests exist, but seem to clash with the firewall.
+    doCheck = false;
+
+    preBuild = ''
+      export GO15VENDOREXPERIMENT=1
```

Comment:
if you're enabling this, I think it means all the golang buildInputs are ignored in favor of the vendor/ directory in the prometheus tree.  I'm not sure what the current policy (if there is one) on that - but it's probably at least encouraged to use the nixpkgs version of library dependencies if possible.